src/main/java/net/william278/profanitycheckerapi/ProfanityChecker.java

```diff
@@ -91,4 +179,56 @@ public void dispose() throws IllegalStateException {
         close();
     }
 
+    private Set<String> getTextToCheck(String text) {
+        Set<String> toCheck = new HashSet<>();
+        toCheck.add(text);
+        toCheck.addAll(List.of(text.split(" ")));
+        Set<String> temp = new HashSet<>(toCheck);
+        for (String s : temp) {
+            toCheck.add(s.replaceAll("[^a-zA-Z]", ""));
+            toCheck.add(s.replaceAll("[^0-9]", ""));
+            toCheck.add(s.replaceAll("[^a-zA-Z0-9]", ""));
+        }
+        temp = new HashSet<>(toCheck);
+        for (String s : temp) {
+            if (s.length() > 1) {
+                for (int i = 0; i < s.length() - 1; i++) {
+                    toCheck.add(s.substring(0, i) + " " + s.substring(i));
+                }
+            }
+        }
+        temp = new HashSet<>(toCheck);
+        for (String s : temp) {
+            toCheck.addAll(List.of(s.split(" ")));
+        }
+        temp = new HashSet<>(toCheck);
+        for (String s : temp) {
+            toCheck.add(s.replaceAll("0", "o"));
+            toCheck.add(s.replaceAll("1", "i"));
+            toCheck.add(s.replaceAll("2", "z"));
+            toCheck.add(s.replaceAll("3", "e"));
+            toCheck.add(s.replaceAll("4", "a"));
+            toCheck.add(s.replaceAll("5", "s"));
+            toCheck.add(s.replaceAll("6", "g"));
+            toCheck.add(s.replaceAll("7", "t"));
+            toCheck.add(s.replaceAll("8", "b"));
+            toCheck.add(s.replaceAll("9", "g"));
+        }
```

Comment:
Instead of doing this for filtering leetspeak, let's have a static key-value map at the top of the class somewhere mapping numerical `Char`s to their respective alphabetical `Char`s. Then, loop through the entry set and replace.
```java
for (int i = 0; i < word.length; i++) {
	char letter = word[i];
	LEETSPEAK_MAP.entrySet().forEach(entry -> {
		if (letter === entry.getKey()) {
			word[i] = entry.getValue();
		}
	}
}
```
Something like that.